packages/pymedphys_deliverydata/src/pymedphys_deliverydata/dicom/conversion.py

```diff
@@ -29,62 +29,49 @@
 
 import numpy as np
 
-from .core import (
-    DeliveryData, find_relevant_control_points)
+from pymedphys_utilities.transforms import convert_angle_to_bipolar
+from pymedphys_utilities.algorithms import maintain_order_unique
 
+from pymedphys_dicom.rtplan import (
+    get_gantry_angles_from_dicom,
+    build_control_points,
+    replace_fraction_group,
+    replace_beam_sequence,
+    restore_trailing_zeros)
 
-def convert_angle_to_bipolar(angle):
-    angle = np.copy(angle)
-    if np.all(angle == 180):
-        return angle
 
-    angle[angle > 180] = angle[angle > 180] - 360
+from ..base import _DeliveryDataBase
+from ..utilities import (
+    find_relevant_control_points,
+    filter_out_irrelevant_control_points,
+    get_all_masked_delivery_data)
 
-    is_180 = np.where(angle == 180)[0]
-    not_180 = np.where(np.invert(angle == 180))[0]
 
-    where_closest_left_leaning = np.argmin(
-        np.abs(is_180[:, None] - not_180[None, :]), axis=1)
-    where_closest_right_leaning = len(not_180) - 1 - np.argmin(np.abs(
-        is_180[::-1, None] -
-        not_180[None, ::-1]), axis=1)[::-1]
-
-    closest_left_leaning = not_180[where_closest_left_leaning]
-    closest_right_leaning = not_180[where_closest_right_leaning]
+def gantry_tol_from_gantry_angles(gantry_angles):
+    min_diff = np.min(np.diff(sorted(gantry_angles)))
+    gantry_tol = np.min([min_diff / 2 - 0.1, 3])
 
-    assert np.all(
-        np.sign(angle[closest_left_leaning]) ==
-        np.sign(angle[closest_right_leaning])
-    ), "Unable to automatically determine whether angle is 180 or -180"
+    return gantry_tol
 
-    angle[is_180] = np.sign(angle[closest_left_leaning]) * angle[is_180]
 
-    return angle
+def delivery_data_to_dicom(delivery_data: _DeliveryDataBase, dicom_template):
```

Comment:
Not `from_dicom()` and `to_dicom()` methods instead?

Reply:
This does become from_dicom and to_dicom in the top level class.